frontend/src/state/dialog.ts

```diff
@@ -12,102 +11,49 @@ export function createDialogState(editor: EditorState) {
 		visible: false,
 		icon: "" as IconName,
 		heading: "",
-		details: "",
-		buttons: [] as TextButtonWidget[],
+		widgets: defaultWidgetLayout() as WidgetLayout | undefined,
+		/// Necessary because we cannot handle widget callbacks from Rust once the editor instance has panicked.
+		jsComponents: undefined as { details: string; buttons: TextButtonWidget[] } | undefined,
```

Comment:
It sounds like this (and all the related systems involved with it on the JS side) basically exists to just draw the panic handler, and there won't be any others. Can we hardcode more of the layout for the panic dialog to simplify this on the JS/Vue side? Basically, it looks like this is kind of a full *system* when we can really get away just with hard-coding the one dialog. No need to use JSON to define widget layouts or other system things. Less code for a hard-coded dialog that shows on command would reduce the complexity and code maintenance burden. Does it seem feasible to de-systemize this and make it have less code? Or is it already about as simple as possible right now?

Reply:
You could probably do it so that the dialog takes in a generic argument (whatever they are in vue) with the component that it contains, and then fill that either with the rust widgets or with the XML version of the panic widgets.

Reply:
I made this change in a98ef68e4a53eca6ef78f0cc6298f71cd3664553